tests/perf-test/OptimisticReportNames.perf-test.ts

```diff
@@ -12,7 +12,11 @@ import {createRandomReport} from '../utils/collections/reports';
 import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
 
 // Mock dependencies
-jest.mock('@libs/ReportUtils');
+jest.mock('@libs/ReportUtils', () => ({
+    ...jest.requireActual<typeof ReportUtils>('@libs/ReportUtils'),
+    isExpenseReport: jest.fn(),
```

Comment:
Do these need to be mocked because they use the router, and the router is not available in the unit test environment?

I'd love it if all mocks actually included a thorough explanation of why they are necessary.

Reply:
im not an expert on jest but i did a quick check. From my understand the reason we're written it this way is cause we use the actual methods of ReportUtils like `getTitleReportField`. It doesn't use the router so its safe for the test to use.
Additionally we mock some of the methods like isExpenseReport to always true so that setup is perhaps easier.

Also yeah i agree, having this in a comment would be helpful.

Reply:
Thanks for adding a comment, but I don't think it helps me understand why the mock is necessary, or why only these two methods are turned into noops. Chirag's explanation comes close... but I still don't know that I get it.

> Partial mock preserves parseReportRouteParams

Why is this important? In order to try and answer that, I'd need to hunt through a ton of code to try and figure it out. Also, it's not clear what `parseReprotRouteParams` has to do with these methods.

>  and other exports that are used by Report.ts's dependencies

I think that the comment should explain the mocked methods, specifically `isExpenseReport` and `getTitleReportField`, rather than just saying all the other exports are untouched (which is pretty obvious from just looking at the mock).

Reply:
Updated